tests/commands/test_build.py

```diff
@@ -2053,6 +2054,128 @@ def test_builder_with_jujuignore(tmp_path, config):
+        "name": "name-from-metadata",
+        "requires": {
+            "relation-name": {
+                "schema": "http://foo.com/schema-requires.yaml",
+            }
+        },
+        "provides": {
+            "relation-name": {
+                "schema": "http://foo.com/schema-provides.yaml",
+            }
+        },
+        "peers": {
+            "relation-name": {
+                "schema": "http://foo.com/schema-peers.yaml",
+            }
+        },
+    }
+
+    metadata_file = tmp_path / "metadata.yaml"
+    metadata_file.write_text(yaml.safe_dump(metadata_data))
+
+    # simple source code
+    src_dir = tmp_path / "src"
+    src_dir.mkdir()
+    charm_script = src_dir / "charm.py"
+    charm_script.touch()
+
+    monkeypatch.chdir(tmp_path)  # so the zip file is left in the temp dir
+    builder = Builder(
+        {
+            "from": tmp_path,
+            "entrypoint": charm_script,
+            "requirement": [],
+        },
+        config,
+    )
+
+    monkeypatch.setenv("CHARMCRAFT_MANAGED_MODE", "1")
+    with patch(
+        "charmcraft.commands.build.check_if_base_matches_host",
+        return_value=(True, None),
+    ):
+        zipname = builder.run()[0]
+
+    zf = zipfile.ZipFile(zipname)
+    assert yaml.safe_load(zf.read("metadata.yaml")) == {
+        "name": "name-from-metadata",
+        "requires": {
+            "relation-name": {
+                "schema": {"type": "requires"},
+            }
+        },
+        "provides": {
+            "relation-name": {
+                "schema": {"type": "provides"},
+            }
+        },
+        "peers": {
+            "relation-name": {
+                "schema": {"type": "peers"},
+            }
+        },
+    }
+
+
+def test_build_invalid_schema(tmp_path, monkeypatch, config):
+    """Integration test: error is raised for invalid schema URL."""
+    build_dir = tmp_path / BUILD_DIRNAME
+    build_dir.mkdir()
+
+    # the metadata (save it and restore to later check)
+    metadata_data = {
+        "name": "name-from-metadata",
+        "requires": {
+            "relation-name": {
+                "schema": "bad-url",
+            }
+        },
+    }
+
+    metadata_file = tmp_path / "metadata.yaml"
+    metadata_file.write_text(yaml.safe_dump(metadata_data))
+
+    # simple source code
+    src_dir = tmp_path / "src"
+    src_dir.mkdir()
+    charm_script = src_dir / "charm.py"
+    charm_script.touch()
+
+    monkeypatch.chdir(tmp_path)  # so the zip file is left in the temp dir
+    builder = Builder(
+        {
+            "from": tmp_path,
+            "entrypoint": charm_script,
+            "requirement": [],
+        },
+        config,
+    )
+
+    monkeypatch.setenv("CHARMCRAFT_MANAGED_MODE", "1")
+    with pytest.raises(CommandError, match="Couldn't fetch schema `bad-url`"), patch(
+        "charmcraft.commands.build.check_if_base_matches_host",
+        return_value=(True, None),
+    ):
+        builder.run()[0]
+
+
```

Comment:
Consider adding a test where schema value is not a string to be interpreted as a URL.